sdk/schemaregistry/azure-schemaregistry-client/src/main/java/com/azure/schemaregistry/client/CachedSchemaRegistryClientBuilder.java

```diff
@@ -0,0 +1,284 @@
+     * This method can be used by passing in a method reference, e.g. ByteEncoder::parseSchemaString.
+     *
+     * The parseMethod argument should be a stateless, idempotent function.
+     *
+     * @param schemaType schema type for which the parse method should be applied.
+     * @param parseMethod function for deserializing registry-stored schema strings to Java Objects
+     * @return The updated {@link CachedSchemaRegistryClientBuilder} object.
+     */
+    public CachedSchemaRegistryClientBuilder loadSchemaParser(
+        String schemaType, Function<String, Object> parseMethod) {
+        if (schemaType == null || schemaType.isEmpty()) {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("Serialization type cannot be null or empty."));
+        }
+        if (this.typeParserDictionary.containsKey(schemaType.toLowerCase(Locale.ENGLISH))) {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("Multiple parse methods for single serialization type may not be added."));
+        }
+        this.typeParserDictionary.put(schemaType.toLowerCase(Locale.ENGLISH), parseMethod);
+        return this;
+    }
+
+    /**
+     * Creates a {@link CachedSchemaRegistryClient} based on options set in the builder.
+     * Every time {@code buildClient()} is called a new instance of {@link CachedSchemaRegistryClient} is created.
+     *
+     * If {@link #pipeline(HttpPipeline) pipeline} is set, then all HTTP pipeline related settings are ignored
+     * endpoint} are when creating the {@link CachedSchemaRegistryClient client}.
+     *
+     * @return A {@link CachedSchemaRegistryClient} with the options set from the builder.
+     * @throws NullPointerException if parameters are incorrectly set.
+     * @throws IllegalArgumentException if credential is not set.
+     */
+    public CachedSchemaRegistryClient buildClient() {
+        HttpPipeline pipeline = this.httpPipeline;
+        // Create a default Pipeline if it is not given
+        if (pipeline == null) {
+            // Closest to API goes first, closest to wire goes last.
+            final List<HttpPipelinePolicy> policies = new ArrayList<>();
+
+            policies.add(new UserAgentPolicy(httpLogOptions.getApplicationId(), clientName, clientVersion,
+                Configuration.getGlobalConfiguration().clone()));
+            policies.add(new RequestIdPolicy());
+            policies.add(new AddHeadersPolicy(this.headers));
+
+            HttpPolicyProviders.addBeforeRetryPolicies(policies);
+
+            policies.add(retryPolicy);
+
+            policies.add(new AddDatePolicy());
+            // Authentications
+            if (credential != null) {
+                // User token based policy
+                policies.add(new BearerTokenAuthenticationPolicy(credential, DEFAULT_SCOPE));
+            } else {
+                // Throw exception that credential and tokenCredential cannot be null
```

Comment:
Let's check for this when starting the construction of the `HttpPipeline`, no need to waste time to just throw.